kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt

```diff
@@ -329,23 +346,11 @@ class TunnelService : VpnService() {
     }
 
     private fun updateStatusNotification(message: String?) {
```

Comment:
@jasonboukheir So I think maybe this might not be as trivial as first thought. When connlib receives the disconnect from the portal, it'll bubble up `onDisconnect(error)` here. That will close this service, removing the notification.

Maybe this UI notification is not needed on Android, since the user will know that Firezone isn't connected anymore since there won't be a "Firezone Connected" perma notification.

Reply:
Scratch that, I think we'll need some kind of notification on Android.

How we do it on Apple is we write the shutdown reason to a file that the UI reads whenever there's a tunnel state change. Then, if it was due to `auth canceled` it shows a notification.

We could do that on Android, or we could try to make sure the notification sent from the TunnelService stays around after the service stops.